### DISCUSSION TITLE
Variables as a color?

### DISCUSSION THREAD

**Discussion by mxcdh** [tailwindlabs/tailwindcss#8041] [Help]
I know that tailwind default, not support variable as parameter CSS like color etc. Are there any hacks on the market for this?

My code:
![image](https://user-images.githubusercontent.com/98115983/161549770-1c0fea6a-c6b5-4c93-82dc-4e8ba17fe3c5.png)

**Comment by SSHari** [comment]
Yeah. So based on the following: https://tailwindcss.com/docs/content-configuration#class-detection-in-depth, tailwind can't detect CSS names which are created dynamically. I assume the problem is that the CSS you need isn't generated. If that's the case, you could probably just store the classes you want in an object and just look them up.

Something like:

```tsx
const disclosureColorStyles = {
	error: 'text-error bg-error/60'
}

<Disclosure.Button className={`text-left ${disclosureColorStyles[color]}`}
```

**Note:** As for the `bg-${color}/60`, if those represent custom colors then I don't think tailwind will create those styles for you automatically. I also don't see them listed in your `theme` in your `tailwind.config.js` file, so that may be another source of issues.

**Reply by mxcdh** [reply]
@SSHari  below my code probably is better, what you think?
At the moment my fast hack:
```
const DisclureCustom = ({ title, color = "metal", children }: Props) => {
  const switchColor = ` text-${color} bg-${color}/60 hover:bg-${color}/25 focus-visible:ring-${color}`;

  return (
    <Disclosure>
      {({ open }) => (
        <div className="pt-2">
          <Disclosure.Button
            className={
              switchColor +
              `flex justify-start w-full px-4 py-2  rounded-lg  focus:outline-none focus-visible:ring  focus-visible:ring-opacity-75`
            }
          > ...
    </Disclosure>
  );
};

export default DisclureCustom;
```

**Reply by SSHari** [reply]
I think you run into the same problem. It's not a question of where you define the styles (i.e. `switchColor` variable vs in the `className` prop), but more so that tailwind won't be able to tell which colors to create based on `text-${color}` because that value is loaded at run time.

Now if you know that you always want to allow all those colors to be available, you could probably force tailwind to create these styles using a safelist in your config file: https://tailwindcss.com/docs/content-configuration#safelisting-classes.

You can also use a regular expression pattern to make it easier to maintain:
```js
// tailwind.config.js
module.exports = {
  content: [],
  safelist: [
    {
	  // you can add the rest of your colors here
      pattern: /text-(error|metal|midnight)/,
    }
  ],
  // ...
}
```

**Reply by gedewon** [reply]
Hi @SSHari 
same issue here
I tried the look-up options but it's still not working.it will inject it after render (after tailwind class scan ) but can't see that actual effect on the page at run time .

**Reply by SSHari** [reply]
@Gedewon would you be able to share the following so that we could get a better idea of your environment:
- `tailwind.config.js`
- code snippet with your styles (and file path)
- Your input.css for the tailwind command

**Reply by gedewon** [reply]
@SSHari  Thank you for your reply!!

### productsColor.json file structure

![image](https://user-images.githubusercontent.com/56429354/162445830-869ea893-f0bd-406e-90a4-8e58b1070355.png)

this is how I require it on the file 

![image](https://user-images.githubusercontent.com/56429354/162445683-57ed1495-4c7f-40f7-807b-834dbc6422e4.png)

this is how it's injected into the dom element

![image](https://user-images.githubusercontent.com/56429354/162446178-459e53cc-dde8-4d28-aa67-e85ae23daa62.png)

Tailwind config

```js
module.exports = {
  // mode:'jit',
  content: ["./pages/**/*.{js,ts,jsx,tsx}", "./src/**/*.{js,ts,jsx,tsx}"],
  theme: {
    extend: {
      colors: {
        background: "#FDFEFF",
        backgroundLight: "#ECEFF9",
        dark: "#052746",
        primary: "#134CA4",
        light: "#2AA7E0",
        emailbox: "#9AADE1",
        Petrol: "bg-gradient-to-br from-orange-300 via-rose-500 to-rose-900",

      },
      fontFamily: {
        primary: ["Nunito", "sans-serif"],
      },
      width:{
        '1/3.5':'36.333333%'
      }

    },
  },
  plugins: [require("@tailwindcss/forms"), require("@tailwindcss/aspect-ratio")],
};

```

**Reply by SSHari** [reply]
@Gedewon Ah ok cool. So it looks like the CSS you’re trying to reference is in that json file.

Now despite that you’re importing it in a `jsx` (or `tsx`) file, I don’t think tailwind will follow import statements and scan the json file.

Could you try to explicitly add that json file path to the content array in your `tailwind.config.js` file?

**Reply by gedewon** [reply]
@SSHari ya it seems nice but not working for me for some reason?

**Reply by SSHari** [reply]
@Gedewon how are you building your tailwind styles? cli, postcss? Are you using a framework?

**Reply by gedewon** [reply]
framework

**Reply by gsw85** [reply]
The quick fix work around for this one is that, you need to prefix all color you want to generated in the one element and set that element to sr-only,  add this element anywhere in your code, for example, in your case:
`<div className="sr-only text-red bg-red text-blue bg-blue text-orange bg-orange">Pre-generated Color </div>`

so above the Tailwind will prebuild the those CSS and able to load it correctly later

**Comment by superuserpro** [comment]
https://qanswer.space/questions/incorporate-unique-color-variables-from-the-tailwindconfigjs-file
